p2p/protocol/identify/id.go

```diff
@@ -252,9 +252,9 @@ func (ids *IDService) consumeMessage(mes *pb.Identify, c inet.Conn) {
 	ids.addrMu.Lock()
 	switch ids.Host.Network().Connectedness(p) {
 	case inet.Connected:
-		ids.Host.Peerstore().AddAddrs(p, lmaddrs, pstore.ConnectedAddrTTL)
+		ids.Host.Peerstore().SetAddrs(p, lmaddrs, pstore.ConnectedAddrTTL)
```

Comment:
This function is really deceptive. It sets the TTLs for the given addrs but doesn't actually swap out the old addrs for a new set.

As far as I can tell, nobody actually _uses_ this. We now use `UpdateAddrs` to atomically change TTLs.

Reply:
I don't think `UpdateAddrs` is sufficient here: it only updates the given addrs and requires us to know the old TTL.
Specifically, if the most recent `IdentifyPush` _removed_ addrs from the set, ie maybe because autorelay has rewritten the addr set, then the old addrs not in the new set will stick around forever!

Reply:
My proposal is to modify `SetAddrs` to discard the old addrs and act as a real setter (and be its first user), and introduce a new `GetAddrs` method that allows us to retrieve the old addrs associated with some TTL. The latter will allow us to avoid clobbering permanent Addrs.

Reply:
Actually, there might be a way to do the dance with `UpdateAddrs` and `AddAddrs`. 
First update addrs of `ConnectedTTL` to `TempTTL` and then `AddAddrs` with connected ttl.

It doesn't have the same effect as removing addrs, but it's close as it will mark them eligible for garbage collection and reduce the exposure time for gossiping invalid addrs.

Reply:
We can go one step further and set the TTL to 0 to invalidate them, no interface change necessary.

Reply:
Fixed and rebased/squased//force-pushed.